### PR TITLE
add parser support for drop/create functions

### PR DESCRIPTION
**Pull request by kovrus**

### CODE REVIEW

File: sql-parser/src/main/java/io/crate/sql/tree/AstVisitor.java [hunk 1]
```diff
@@ -504,4 +516,5 @@ public R visitBegin(BeginStatement node, C context) {
     public R visitShowTransaction(ShowTransaction showTransaction, C context) {
         return visitStatement(showTransaction, context);
     }
+
```
Comment (bogensberger) [inline sql-parser/src/main/java/io/crate/sql/tree/AstVisitor.java]: remove

File: sql-parser/src/main/java/io/crate/sql/tree/CreateFunction.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+import java.util.Objects;
+import java.util.Set;
+
+public class CreateFunction extends Statement {
+
+    public enum FunctionLanguage {
+        JAVASCRIPT
+    }
+
+    public enum Option {
```
Comment (bogensberger) [inline sql-parser/src/main/java/io/crate/sql/tree/CreateFunction.java]: pls rename it (maybe mode?)

File: sql-parser/src/main/java/io/crate/sql/tree/CreateFunction.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(name, replace, arguments, returnType, language, options);
+    }
+
+    @Override
+    public String toString() {
+        return MoreObjects.toStringHelper(this)
```
Comment (mfussenegger) [inline sql-parser/src/main/java/io/crate/sql/tree/CreateFunction.java]: For the future: I'd prefer that we use the other toString template that doesn't depend on guava

Comment (kovrus) [inline sql-parser/src/main/java/io/crate/sql/tree/CreateFunction.java]: @mfussenegger sure, will change that

**Pull request merged**